src/main/java/homura/Homura.java

```diff
@@ -0,0 +1,356 @@
+     *
+     * @param inp The full line of input to the bot.
+     */
+    public static void cmdDelete(String inp) {
+        String[] splitInps = inp.split(" ");
+        int i = Integer.parseInt(splitInps[1]) - 1;
+        if (i >= todos.size()) {
+            throw new InvalidInputHomuraException("delete", inp);
+        }
+        Todo t = todos.get(i);
+        todos.remove(i);
+        System.out.println(
+                INDENT + DIVIDER + '\n'
+                + INDENT + " " + t.getClass().getSimpleName()
+                + " removed" + '\n'
+                + INDENT + INDENT + t + '\n'
+                + INDENT + " " + todos.size()
+                + " tasks(s) in your list" + '\n'
+                + INDENT + DIVIDER + '\n'
+        );
+    }
+    /**
+     * Handles the logic of the find command.
+     *
+     * @param inp The full line of input to the bot.
+     */
+    public static void cmdFind(String inp) {
+        String[] splitInps = inp.split(" ");
+        String s= splitInps[1];
+        ArrayList<Todo> matches = todos.findTodosWith(s);
+        Ui.printFoundTodosFormatted(matches);
+    }
+
+
+
+
+
+    // Main ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    public static void main(String[] args) {
+        on();
+
+        while (true) {
+            // How to read input inspired by
+            // https://github.com/Bryce-3D/My-Codeforces-Codes/blob/main/
+            // Java/0001-0100/CF_0001A.java
+            String inp = SC.nextLine();
+
+            // Exit the chatbot
+            if (inp.equals("bye")) {
+                off();
+                break;
+            }
+
+            // List items in the Todo list
+            if (inp.equals("list")) {
+                Ui.printTodosFormatted();
+                continue;
+            }
+
+            String[] splitInps = inp.split(" ");
+            String cmd = splitInps[0].toLowerCase();
+
+            switch (cmd) {
+            case "mark":   // Mark a Todo on the list
+                cmdMark(inp);
+                break;
+            case "unmark":   // Unmark a Todo on the list
+                cmdUnmark(inp);
+                break;
+            case "todo":   // Add a Todo to the list
+                cmdTodo(inp);
+                break;
+            case "deadline":   // Add a deadline to the list
+                cmdDeadline(inp);
+                break;
+            case "event":   // Add an event to the list
+                cmdEvent(inp);
+                break;
+            case "delete":   // Remove an event from the list
+                cmdDelete(inp);
+                break;
+            case "find":   // Find todos in the list with some text
+                cmdFind(inp);
+                break;
+            default:   // Not a command
+                throw new InvalidCmdHomuraException(cmd);
+            }
+        }
+    }
+}
+
+
+
+
+
+// RECYCLING BIN ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
How about moving this "recycling bin" to another file, and keeping this original file tidy? Although I am not sure if it is a coding standard violation.